### PR TITLE
Fix for runtime-live-build break per Santi's suggestion

### LINKED ISSUE #31985: runtime-live-build failing on all PRs
From @safern:
"I believe it’s the cmake error."
"We've got to pass -warnAsError:0 on the Windows legs"
"Linux is not affected"
https://github.com/dotnet/runtime/blob/master/eng/pipelines/common/global-build-job.yml#L40
"You just need to define an intermediate variable and set it to empty and then to the arg if platform is Windows_NT. Because the syntax is different for bash and PS and this only fails on Windows"

cc: @ViktorHofer, @jkoritzinsky, @trylek

### PR DESCRIPTION
**Pull request by trylek**
Fixes: https://github.com/dotnet/runtime/issues/31985

### CODE REVIEW

**Comment by trylek** [comment]
OK, I believe we're past the problematic point so the fix should be good to go.

**Review by stephentoub — APPROVED** [review]
Thanks! If this gets CI green, let's get it merged, and any feedback can be addressed subsequently.

File: eng/pipelines/common/global-build-job.yml [hunk 1]
```diff
@@ -19,10 +19,18 @@ jobs:
     pool: ${{ parameters.pool }}
     container: ${{ parameters.container }}
     timeoutInMinutes: ${{ parameters.timeoutInMinutes }}
-    variables: ${{ parameters.variables }}
     workspace:
       clean: all
 
+    variables:
+    - ${{ each variable in parameters.variables }}:
+      - ${{insert}}: ${{ variable }}
```
Comment (safern) [inline eng/pipelines/common/global-build-job.yml]: Nit: you don’t need the for each here, you can get this done by inserting the whole object:

```yml
${{ insert }}: ${{ parameters.variables }}
```

**Review by safern — COMMENTED** [review]
Thanks for fixing @trylek. LGTM

**Pull request merged**